app/controllers/SchemeDetailsController.scala

```diff
@@ -29,35 +28,32 @@ import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import utils.DateHelper
 import views.html.schemeDetails
 
+import scala.concurrent.Future
+
 class SchemeDetailsController @Inject()(appConfig: FrontendAppConfig,
-                                          override val messagesApi: MessagesApi,
-                                          schemeDetailsConnector: SchemeDetailsConnector,
-                                          listSchemesConnector: ListOfSchemesConnector,
-                                          authenticate: AuthAction,
-                                          getData: DataRetrievalAction,
+                                        override val messagesApi: MessagesApi,
+                                        schemeDetailsConnector: SchemeDetailsConnector,
+                                        listSchemesConnector: ListOfSchemesConnector,
+                                        authenticate: AuthAction,
+                                        getData: DataRetrievalAction,
                                         userAnswersCacheConnector: UserAnswersCacheConnector
                                        ) extends FrontendController with I18nSupport {
 
-  def onPageLoad(srn: String): Action[AnyContent] = (authenticate andThen getData).async {
+  def onPageLoad(srn: String): Action[AnyContent] = authenticate.async {
     implicit request =>
 
       schemeDetailsConnector.getSchemeDetails("srn", srn).flatMap { scheme =>
         listSchemesConnector.getListOfSchemes(request.psaId.id).flatMap { list =>
           val schemeDetail = scheme.schemeDetails
-          val minimalSchemeDetails = MinimalSchemeDetail(srn, schemeDetail.pstr, schemeDetail.name)
-
-          userAnswersCacheConnector.save(request.externalId, MinimalSchemeDetailId, minimalSchemeDetails).map { _ =>
-
-            val isSchemeOpen = schemeDetail.status.equalsIgnoreCase("open")
+          val isSchemeOpen = schemeDetail.status.equalsIgnoreCase("open")
```

Comment:
Not sure about this but it might be nicer to have the "open" string externalised to one central place - config file maybe or else a central constants file??

Reply:
this is an existing code but i think this should be an enum may be for all the different scheme status. we might need to revisit this and refactor it later, as these are being used all over the place like a hardcoded string.

Reply:
Fair enough. Maybe create a Jira ticket in backlog to look at this later if we have time.